packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -3646,6 +3646,51 @@ describeWithDOM('mount', () => {
         expect(wrapper.prop('bar')).to.equal('bye');
       });
     });
+
+    describe('props in async handler', () => {
+      class TestComponent extends React.Component {
+        constructor(props) {
+          super(props);
+          this.state = { counter: 1 };
+          this.handleClick = this.handleClick.bind(this);
+          this.delay = this.delay.bind(this);
+        }
+
+        async handleClick() {
+          this.delay(100);
+          return new Promise((resolve) => {
+            this.setState({ counter: 2 }, () => {
+              resolve();
+            });
+          });
+        }
+
+        delay(ms) {
```

Comment:
this can just be a standalone function; it doesn't need to be an instance method.